pkg/lib/authn/sso/userinfo.go

```diff
@@ -61,6 +63,16 @@ func DecodeAzureADv2(userInfo map[string]interface{}) *ProviderUserInfo {
 	}
 }
 
+func DecodeADFS(userInfo map[string]interface{}) *ProviderUserInfo {
+	id, _ := userInfo["sub"].(string)
+	email, _ := userInfo["email"].(string)
+
+	return &ProviderUserInfo{
+		ID:    id,
+		Email: email,
+	}
```

Comment:
I found that not all our oauth provider is calling `DecodeUserInfo`? `DecodeUserInfo` is useful only if we call it in `<provider>.go`? If this is the case, maybe we should not add this, since this is not the way to convert `userInfo` to `ProviderUserInfo` for adfs, it is confusing.